ambari-server/src/main/java/org/apache/ambari/server/checks/DatabaseConsistencyCheckHelper.java

```diff
@@ -810,6 +811,49 @@ static void checkMySQLEngine () {
     }
   }
 
+  /**
+   * This method checks for stale alert definitions..
+   * */
+  static void checkForStalealertdefs () {
+    Configuration conf = injector.getInstance(Configuration.class);
+
+    LOG.info("Checking to ensure there is no stale alert definitions");
+
+    ensureConnection();
+
+    String STALE_ALERT_DEFINITIONS = "select definition_name, service_name from alert_definition where service_name not in (select service_name from clusterservices) and service_name not in ('AMBARI')";
+
+    ResultSet rs = null;
+    Statement statement;
+
+    try {
+      statement = connection.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE);
+      rs = statement.executeQuery(STALE_ALERT_DEFINITIONS);
+      if (rs != null) {
+        Map<String, String> alertInfo = new HashMap<>();
+        while (rs.next()) {
+          alertInfo.put(rs.getString("definition_name"),rs.getString("service_name"));
+        }
+        if (!alertInfo.isEmpty()){
+          String alertInfoStr = "";
+          for (Map.Entry<String, String> entry : alertInfo.entrySet()) {
+            alertInfoStr = entry.getKey() + "(" + entry.getValue() + ")" ;
+          }
+          warning("You have Alerts that are not mapped with any services : {}", alertInfoStr);
```

Comment:
We should also add an autofix to just delete them, so this warning can also say run autofix to fix them.

Reply:
@swagle as part of auto-fix, have to take care of deleting all references like history, current as well. I would prefer doing that as part of another BUG. please let me know.